config/store.go

```diff
@@ -248,9 +248,28 @@ func (s *Store) loadLockedWithOld(oldCfg *model.Config, unlockOnce *sync.Once) e
 	if err != nil {
 		return errors.Wrap(err, "failed to marshal loaded config")
 	}
-	if !s.readOnly && (len(configBytes) == 0 || !bytes.Equal(oldCfgBytes, newCfgBytes)) {
-		if err := s.backingStore.Set(s.configNoEnv); err != nil {
-			if !errors.Is(err, ErrReadOnlyConfiguration) {
+
+	var shouldStore bool
+	hasChanged := len(configBytes) == 0 || !bytes.Equal(oldCfgBytes, newCfgBytes)
+	if hasChanged {
+		featureFlags := s.configNoEnv.FeatureFlags
+		// Don't persist feature flags unless we are on MM cloud
+		// MM cloud uses config in the DB as a cache of the feature flag
+		// settings in case the management system is down when a pod starts.
+		if !s.persistFeatureFlags {
+			s.configNoEnv.FeatureFlags = nil
+		}
```

Comment:
Should we refactor this to a separate method? Seems to be repeated in the `Get` method too.

Reply:
You mean in `Set()`? I've purposely repeated but I guess we can make it method since it's all pointers.

Reply:
0/5. Since the comment was repeated, it seemed like an important piece of logic that shouldn't be missed out if this needs to be done elsewhere again. But can be part of the cleanup.